lib/ruby_lsp/type_inferrer.rb

```diff
@@ -39,6 +39,26 @@ def infer_receiver_for_call_node(node, node_context)
       case receiver
       when Prism::SelfNode, nil
         self_receiver_handling(node_context)
+      when Prism::StringNode
```

Comment:
I think you can also handle ranges and maybe procs

Reply:
Also percent-literals (`%w{}` etc)?

Reply:
Added ranges and literal lambdas. `%{}` are string nodes, so they are already handled.